core/jobs/types/base_validation_errors.py

```
@@ -49,11 +64,23 @@ def __init__(self, message, model_or_kind, model_id=None):
             raise ValueError('message must be a non-empty string')
 
         if model_id is None:
+            # Here, get_model_id and get_model_kind can only accept argument
+            # of type base_models.BaseModel but model_or_kind is defined as a
+            # type of Union[base_models.BaseModel, str]. So, to rule out the
+            # possibility of str of model_or_kind for mypy type checking.
+            # We asserted it as an Basemodel type.
+            assert isinstance(model_or_kind, base_models.BaseModel)
             model_id = job_utils.get_model_id(model_or_kind)
             model_kind = job_utils.get_model_kind(model_or_kind)
         else:
-            model_kind = model_or_kind
-
+            # The type of model_or_kind is Union[base_models.BaseModel, str]
+            # and the type of model_kind is str, due to difference in type MyPy
+            # throws an assignment error. Thus to silent an error, we added an
+            # ignore here.
+            model_kind = model_or_kind  # type: ignore[assignment]
```

Should we ignore this error?
doesn't adding `assert isinstance(model_or_kind, str)` above this line prevent the error?

---

Yes, you are right!
Done!